### ISSUE TITLE
Standard Scripts

### ISSUE THREAD

**Issue by ianmilligan1** [archivesunleashed/auk#2]
Each collection will have a set of standard scripts run when first ingested.

**Comment by ianmilligan1** [comment]
FWIW I don't know if this is helpful, but here's a version of the standard derivative script I use to test AUT - generates what we need:

```
import io.archivesunleashed.spark.matchbox.{ExtractDomain, ExtractLinks, RemoveHTML, RecordLoader, WriteGEXF}
import io.archivesunleashed.spark.rdd.RecordRDD._
val r = RecordLoader.loadArchives("/data/govt-info/*.gz", sc).keepValidPages().map(r => ExtractDomain(r.getUrl)).countItems().saveAsTextFile("/data/all-domains")
RecordLoader.loadArchives("/data/govt-info/*.gz", sc).keepValidPages().map(r => (r.getCrawlDate, r.getDomain, r.getUrl, RemoveHTML(r.getContentString))).saveAsTextFile("/data/all-text")
val links = RecordLoader.loadArchives("/data/govt-info/*.gz", sc).keepValidPages().map(r => (r.getCrawlDate, ExtractLinks(r.getUrl, r.getContentString))).flatMap(r => r._2.map(f => (r._1, ExtractDomain(f._1).replaceAll("^\\s*www\\.", ""), ExtractDomain(f._2).replaceAll("^\\s*www\\.", "")))).filter(r => r._2 != "" && r._3 != "").countItems().filter(r => r._2 > 5)
WriteGEXF(links, "/data/links-for-gephi.gexf")
sys.exit
```

executes when passed to Scala. The paths obviously need to be changed but they can be passed with variables (when I do mass testing on `indexer` I'll script it out).

**Comment by ianmilligan1** [comment]
Closed with  1e43a20!
